### ISSUE TITLE
A question about the parameter cluster_file in generate_chain_data_cache.py.

### ISSUE THREAD

**Issue by XXXXXcoder** [aqlaboratory/openfold#82]
![image](https://user-images.githubusercontent.com/35370756/153744405-96715eb5-c7eb-472e-86f8-4d8196f9c1a7.png)
"clusters-by-entity-40.txt" as the parameter cluster_file is required by generate_chain_data_cache.py. But I don't find this file in my path, so I want to know how the file "clusters-by-entity-40.txt" generates.

Thanks so much!

**Comment by gahdritz** [comment]
The cluster file isn't downloaded by default by the installation script, as the cluster file you'll want to use depends on your choice of database. For the default AlphaFold databases, you'll want to download the cluster file linked in the screenshot you sent. Hope this clears things up.

**Comment by XXXXXcoder** [comment]
Thanks so much! There is another question:
I use the same procedure DeepMind used by running "python3 scripts/precompute_alignments.py" for generating MSAs, but this procedure computes MSAs of ~180,000 protein chain, takes up a lot of time and computing resources, so can I just compute ~400 MSAs so that I just run the code successfully regardless of model effects?

**Comment by gahdritz** [comment]
Just move 400 mmCIF files to a new directory and run the script on those.

**Comment by XXXXXcoder** [comment]
![image](https://user-images.githubusercontent.com/35370756/154430962-d5961e76-c021-49c8-905f-95c487ca0e07.png)

Thanks for your answer! There is still a question that I need your help: Is the value of first path parameter "mmcif_dir/" the same as the value of third path parameter "template_mmcif_dir/"?

**Comment by gahdritz** [comment]
`mmcif_dir` contains training data in the form of .cif mmCIF files or PDB files with all of the columns output by OpenFold. `template_mmcif_dir` contains mmCIF files for template chains. Sometimes, these can be the same (for the first phase of AlphaFold 2 training, for example).

**Comment by XXXXXcoder** [comment]
Thanks! Besides, can this code about the OpenFold run with reference to readme.txt? Because "train_openfold.py" requires a parameters "max_template_date", but it doesn't set by readme.txt, as shown in the figure.
![image](https://user-images.githubusercontent.com/35370756/155059168-69e3b72e-1ad3-4b4d-b7fc-72fa8292de06.png)

**Comment by gahdritz** [comment]
It's a positional parameter on the second line ("2021-10-10")

**Comment by XXXXXcoder** [comment]
Yep...sorry...I am careless... There is a new question: I have computed 370 alignments, which covered 151 proteins, but when I run the "generate_chain_data_cache.py", I got a output file"chain_data_cache.json" whose content is "{}", so when I run the "train_openfold.py", I got an error:"chain_data_cache_entry = chain_data_cache[chain_id]    KeyError: '5g2e_A'". How can I solve this error? Do you have any suggestions?
![image](https://user-images.githubusercontent.com/35370756/155065509-a7e54a04-324b-406e-9888-0f7d5719fcbf.png)

**Comment by gahdritz** [comment]
I just pushed a commit that changed the default behavior of the chain data cache file. Previously, chains that didn't appear in the chain cluster file were excluded entirely. Now, they're set to -1.

**Comment by XXXXXcoder** [comment]
![image](https://user-images.githubusercontent.com/35370756/156114828-daa517c0-fb71-4541-9f1d-a5939f9b8167.png)

![image](https://user-images.githubusercontent.com/35370756/156114466-0e90893c-6cd9-4c47-b1fd-6f90b61aafdc.png)
the base class of "OpenFoldWrapper" is "LightningModule", which also has no attribute '_compute_validation_metrics'. Is this a bug?

**Comment by gahdritz** [comment]
This has since been fixed. Try pulling one more time.

**Comment by XXXXXcoder** [comment]
Thanks！but there is a new error:
![image](https://user-images.githubusercontent.com/35370756/156358373-6a113839-bb48-42a1-a36b-9bd56f309a3d.png)
should I change the parameters "num_workers" in config.py? as shown in following figure:
![image](https://user-images.githubusercontent.com/35370756/156358947-8377376e-47cb-4581-af58-8704ba05f1e1.png)

**Comment by gahdritz** [comment]
That's odd---I've never seen this one before. Could you share more information about what's causing it? Possibly share minimal reproduction code?

**Comment by XXXXXcoder** [comment]
![image](https://user-images.githubusercontent.com/35370756/156502764-6b3290c5-3b46-4ee6-8de4-8348b2f8b524.png)
I don't change your code, just run the code based on your readme.txt

**Comment by XXXXXcoder** [comment]
trainging data is 370 alignments, which covered 151 proteins. Error information is as follows:
[error information.txt](https://github.com/aqlaboratory/openfold/files/8175238/error.information.txt)

**Comment by gahdritz** [comment]
I tried reproducing this earlier but it worked normally. What versions of each package are you running?

**Comment by XXXXXcoder** [comment]
[package version.txt](https://github.com/aqlaboratory/openfold/files/8175249/package.version.txt)
package version

**Comment by XXXXXcoder** [comment]
I have resolved this problem via insert a line of code "torch.multiprocessing.set_sharing_strategy('file_system')"

**Comment by XXXXXcoder** [comment]
But I get a same prolem as #50, i.e., loss is NaN.

**Comment by mooninrain** [comment]
I can't see the download link for PDB40 in current main branch. I found one here, though:
https://www.rcsb.org/docs/programmatic-access/file-download-services
